File: 500/554 - Brick Wall.java
import java.util.HashMap;

class Solution {
    public int leastBricks(List<List<Integer>> wall) {
        HashMap<Integer, Integer> hm = new HashMap<Integer, Integer>();
        int m = 0, pos, i;
        
        for (List<Integer> l : wall) {
            pos = 0;
            
            for (i = 0; i < l.size() - 1; ++i) {
                pos += l.get(i);
                if (hm.containsKey(pos))
                    hm.replace(pos, hm.get(pos) + 1);
                else
                    hm.put(pos, 1);
                
                m = Math.max(m, hm.get(pos));
            }
        }
        
        return wall.size() - m;
    }
}

File: csharp/554.brick-wall.cs
/*
 * @lc app=leetcode id=554 lang=csharp
 *
 * [554] Brick Wall
 */

using System.Linq;
using System.Collections.Generic;

// @lc code=start
public class Solution {
    public int LeastBricks(IList<IList<int>> wall) {
        IDictionary<int, int> map = new Dictionary<int, int>();
        int max = 0;
        foreach (var row in wall) {
            int curr = 0;
            foreach (var brick in row.SkipLast(1)) {
                curr += brick;
                int count = 0;
                map.TryGetValue(curr, out count);
                ++count;
                max = Math.Max(max, count);
                map[curr] = count;
            }
        }
        return wall.Count - max;
    }
}
// @lc code=end

